### PR TITLE
feat(dashboard): Initial pricing domain

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Sets up the initial work for Pricing domain
- Fixes Store domain

**Todo in follow up PR**
- Translations
- Add status when creating the Price List and allow updating it
- Improve DataGrid component
- Add missing functionality once backend support is added (customer_groups, region prices and update prices)

CLOSES CORE-1931

### CODE REVIEW

File: packages/admin-next/dashboard/src/components/grid/grid-cells/non-interactive/display-cell/display-cell.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { clx } from "@medusajs/ui"
+import { PropsWithChildren } from "react"
+
+type DisplayCellProps = PropsWithChildren<{
+  variant?: "base" | "subtle"
+}>
+
+/**
+ * Field for displaying non-editable data in a grid.
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/components/grid/grid-cells/non-interactive/display-cell/display-cell.tsx]: Should we remove it now while we're at it?

File: packages/admin-next/dashboard/src/components/grid/grid-cells/common/currency-cell/currency-cell.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+              data-input-field="true"
+              data-field-id={field}
+              data-cell-type={GridCellType.EDITABLE}
+              className="size-full bg-transparent py-2.5 pr-4 text-right outline-none"
+              style={{
+                paddingLeft: symbolRef.current?.offsetWidth
+                  ? `${symbolRef.current.offsetWidth + 8}px`
+                  : "16px",
+              }}
+              decimalScale={decimalScale} // Temporary until the backend returns the correct number of decimals to use for the currency
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/components/grid/grid-cells/common/currency-cell/currency-cell.tsx]: The BE already returns this now, so we can probably get rid of the helper and rely on that data

File: packages/admin-next/dashboard/src/v2-routes/pricing/pricing-create/components/pricing-create-form/pricing-create-form.tsx [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+      })
+
+      return false
+    }
+
+    return true
+  }
+
+  const isTabDirty = (tab: Tab) => {
+    if (tab === Tab.DETAIL) {
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/v2-routes/pricing/pricing-create/components/pricing-create-form/pricing-create-form.tsx]: NIT: Use a switch statement instead

**Review by sradevski — APPROVED** [review]
I basically skimmed through the PR, but overall it looks fine. Let's get it merged soon and I'll apply this to the products page

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**